### PR TITLE
remove USE_TURBO macro, turbo mode will be provided as an option for …

### PR DESCRIPTION
**Pull request by minshao**
…enabling prefix mode

### CODE REVIEW

File: src/REmatchEngine.cpp [hunk 1]
```diff
@@ -214,13 +205,8 @@ void REmatch2AutomataEngine::compile(const std::vector<Rule>& rules,
...
     mods.push_back(opt);
   }
-  auto matcher =
-      rematch2_compile(ids.data(), exps.data(), mods.data(), ids.size(), reduce
-#ifdef USE_TURBO
-                       ,
-                       turbo
-#endif
-                       );
+  auto matcher = rematch2_compile(ids.data(), exps.data(), mods.data(),
```
Comment (vvalgenti) [inline src/REmatchEngine.cpp]: Shouldn't this be rematch2_compile_with_shortcuts()?

File: src/REmatchEngine.cpp [hunk 2]
```diff
@@ -249,13 +235,8 @@ void REmatch2AutomataEngine::compile_test(const std::vector<Rule>& rules) const
...
     mods.push_back(opt);
   }
-  auto testMatcher =
-      rematch2_compile(ids.data(), exps.data(), mods.data(), ids.size(), reduce
-#ifdef USE_TURBO
-                       ,
-                       turbo
-#endif
-                       );
+  auto testMatcher = rematch2_compile(ids.data(), exps.data(), mods.data(),
```
Comment (vvalgenti) [inline src/REmatchEngine.cpp]: Shouldn't this be rematch2_compile_with_shortcuts()?

File: src/REmatchEngine.cpp [hunk 3]
```diff
@@ -279,13 +260,8 @@ void REmatch2AutomataEngine::update_test(const std::vector<Rule>& rules)
...
     mods.push_back(opt);
   }
-  auto tmp_matcher =
-      rematch2_compile(ids.data(), exps.data(), mods.data(), ids.size(), reduce
-#ifdef USE_TURBO
-                       ,
-                       turbo
-#endif
-                       );
+  auto tmp_matcher = rematch2_compile(ids.data(), exps.data(), mods.data(),
```
Comment (vvalgenti) [inline src/REmatchEngine.cpp]: Shouldn't this be rematch2_compile_with_shortcuts()?

**Review by vvalgenti — CHANGES_REQUESTED** [review]

File: src/REmatchEngine.cpp (hunk 1 above)
Comment (minshao) [reply src/REmatchEngine.cpp]: I am working on it~

**Review by vvalgenti — APPROVED** [review]

**Pull request merged**